model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,369 @@
+// \brief Print the model joints information.
+// \param[in] _entity Entity of the model requested.
+// \param[in] _ecm ECM ready for requests.
+// \param[in] _jointName Joint to be printed, if nullptr, print all joints.
+void printJoints(const uint64_t entity,
+                const ignition::gazebo::EntityComponentManager &_ecm,
+                const std::string &_jointName)
+{
+  static const std::map<sdf::JointType, std::string> jointTypes = {
+    {sdf::JointType::REVOLUTE, "revolute"},
+    {sdf::JointType::BALL, "ball"},
+    {sdf::JointType::CONTINUOUS, "continuous"},
+    {sdf::JointType::FIXED, "fixed"},
+    {sdf::JointType::GEARBOX, "gearbox"},
+    {sdf::JointType::PRISMATIC,  "prismatic"},
+    {sdf::JointType::REVOLUTE2, "revolute2"},
+    {sdf::JointType::SCREW, "screw"},
+    {sdf::JointType::UNIVERSAL, "universal"}
+  };
+
+  const auto joints = _ecm.EntitiesByComponents(
+  ignition::gazebo::components::ParentEntity(entity),
+  ignition::gazebo::components::Joint());
+
+  for(const auto &_entity : joints)
+  {
+    const auto parentComp =
+        _ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+    const auto nameComp =
+        _ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+    if(_jointName.length() && _jointName != nameComp->Data())
+        continue;
+
+    if (parentComp)
+    {
+      std::string parentInfo;
+      auto parentNameComp =
+          _ecm.Component<ignition::gazebo::components::Name>(
+          parentComp->Data());
+
+      if (parentNameComp)
+      {
+        parentInfo += parentNameComp->Data() + " ";
+      }
+      parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+
+      std::cout << "  - Joint [" << _entity << "]" << std::endl
+                << "    - Name: " << nameComp->Data() << std::endl
+                << "    - Parent: " << parentInfo << std::endl;
+    }
+
+    const auto jointComp =
+        _ecm.Component<ignition::gazebo::components::JointType>(_entity);
+    const auto childLinkComp =
+        _ecm.Component<ignition::gazebo::components::ChildLinkName>(_entity);
+    const auto parentLinkComp =
+        _ecm.Component<ignition::gazebo::components::ParentLinkName>(_entity);
+
+    if (childLinkComp && parentLinkComp)
+    {
+      std::cout << "    - Joint type:  " << jointTypes.at(jointComp->Data())
+      << "\n"   << "    - Parent Link: [" << childLinkComp->Data() << "]\n"
+                << "    - Child Link:  [" << parentLinkComp->Data() << "]\n";
+    }
+  }
+}
+}
+
+//////////////////////////////////////////////////
+extern "C" IGNITION_GAZEBO_VISIBLE void cmdModelList()
+{
+  ignition::gazebo::EntityComponentManager ecm{};
+  if(!PopulateECM(ecm))
+  {
+    return;
+  }
+  const auto models = ecm.EntitiesByComponents(
+    ignition::gazebo::components::ParentEntity(1),
+    ignition::gazebo::components::Model());
+
+  std::cout << "Available models:" << std::endl;
+
+  for(const auto &m : models)
```

Comment:
I'm not sure why the code-checker isn't catching these, but it should be complaining about the missing space on several places:

```suggestion
  for (const auto &m : models)
```

Reply:
Addressed